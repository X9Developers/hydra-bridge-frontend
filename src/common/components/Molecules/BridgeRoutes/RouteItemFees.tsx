import React, { useCallback } from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";

import { ContainerCard } from "../../Atoms/Containers/Container";

import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";
import {
  formatGasFees,
  formatServiceTime,
} from "../../../../helpers/formatsHelper";

const StyledText = styled.p`
  color: ${theme.colors.white};
  font-size: ${theme.paragraph.xs};
  margin: 0;
  min-width: 15rem;
  padding: 0.5rem 1.2rem;
`;

export type RouteItemFeesProps = {
  transactionCostInUsd: number;
  serviceTime: number;
};
const RouteItemFees = ({
  transactionCostInUsd,
  serviceTime,
}: RouteItemFeesProps) => {
  const { t } = useTranslation();

  return (
    <ContainerCard
      borderRadius={theme.borderRadius.lg}
      padding={"0"}
      bg={theme.colors.gray["medium-dark"]}
      boxShadow={theme.boxShadow.sm}
    >
      <StyledText data-testid="route-fees">
        {t("gas-fee")} : ~$
        {useCallback(
          () => formatGasFees(transactionCostInUsd),
          [transactionCostInUsd]
        )}{" "}
        | ~{useCallback(() => formatServiceTime(serviceTime), [serviceTime])}{" "}
        {t("min")}
      </StyledText>
    </ContainerCard>
  );
};

export default React.memo(RouteItemFees);
