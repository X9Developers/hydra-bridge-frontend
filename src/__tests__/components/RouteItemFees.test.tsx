import React from "react";
import { cleanup, render } from "@testing-library/react";
import RouteItemFees, {
  RouteItemFeesProps,
} from "../../common/components/Molecules/BridgeRoutes/RouteItemFees";

describe("The Bridge route item fees", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render correctly", async () => {
    const props: RouteItemFeesProps = {
      transactionCostInUsd: 0.39979797399225586,
      serviceTime: 300,
    };

    const { asFragment } = render(<RouteItemFees {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should be defined and correctly calculated", async () => {
    const props: RouteItemFeesProps = {
      transactionCostInUsd: 0.39979797399225586,
      serviceTime: 300,
    };
    const { container } = render(<RouteItemFees {...props} />);
    const element = container.querySelector("p");

    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent("gas-fee : ~$0.40 | ~5 min");
  });

  it("should update correctly", async () => {
    const props: RouteItemFeesProps = {
      transactionCostInUsd: 0.39979797399225586,
      serviceTime: 300,
    };
    const { container, rerender } = render(<RouteItemFees {...props} />);
    const element = container.querySelector("p");

    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent("gas-fee : ~$0.40 | ~5 min");

    rerender(
      <RouteItemFees
        transactionCostInUsd={3.2546656797399225586}
        serviceTime={1800}
      />
    );
    expect(element).toHaveTextContent("gas-fee : ~$3.25 | ~30 min");
  });
});
