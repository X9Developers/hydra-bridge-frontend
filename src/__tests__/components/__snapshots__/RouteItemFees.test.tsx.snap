// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Bridge route item fees should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL sc-jRQBWg fGXmcv fmljMh"
  >
    <p
      class="sc-gKclnd ixACOe"
      data-testid="route-fees"
    >
      gas-fee : ~$0.40 | ~5 min
    </p>
  </div>
</DocumentFragment>
`;
